docs/trace/extending-the-sdk/README.md

```diff
@@ -174,7 +174,10 @@ Writing an instrumentation library typically involves 3 steps.
        method, it should call the `AddInstrumentation` method, and `AddSource`
        method to enable its ActivitySource for the provider. An example
        instrumentation using this approach is [SqlClient
-       instrumentation](../../../src/OpenTelemetry.Instrumentation.SqlClient/TracerProviderBuilderExtensions.cs)
+       instrumentation](../../../src/OpenTelemetry.Instrumentation.SqlClient/TracerProviderBuilderExtensions.cs).
+       **CAUTION**: The instrumentation libraries requiring state management
```

Comment:
@pellared after another pass at this, I am not sure if this is correct. The linked issue does not explicitly state "SqlClient" instrumentation is a hard one to auto-instrument. It only mentions Redis being hard to auto-instrument.

Other instrumentation libraries like Asp.Net Core etc also has state management requirement, but not listed as hard to auto-instrument?

Reply:
Is the difficulty arising from:
1. Library having external references?
2. Because state management, tied to TracerProvider is required?

Reply:
Both 😉

Reply:
I think *all* instrumentation libraries we ship have state management tied to TracerProvider. So what would be our recommendation to users? The extension method should simply do "AddSource()", only...??